### PR TITLE
Spike-on-stick and knife spear work with Sojutsu and Silat

### LINKED ISSUE #32343: knife spear is missing from Sōjutsu fighting style
# see title. simple knife spear is useable with Sōjutsu fightings style but knife spear cant (the new version without the flimsy tag) 

# Steps To Reproduce

1.  pull up the fighting style Sōjutsu
2. read the weapons you can use
3. notice that knife spear inst there but simple knife spear is 

# Versions and configuration

windows 10
0.D-5547-gbf89376 (tiles)

so to fix this someone just needs to add "knife spear" to the list of weapons that can be used with  Sōjutsu fighting style

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Spike on a stick and knife spear work with Sojutsu and Silat"```

#### Purpose of change
Closes #32343. Addition of the improved knife spear and spike on a stick did not coincide with an addition of it to martial arts. 
#### Describe the solution
Addition of weapons to weapons list for Pentjak Silat and Sojutsu.

#### Describe alternatives you've considered
None.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
You might want to also add the spike spear

**Comment by LaVeyanFiend** [comment]
Done.

**Comment by I-am-Erk** [comment]
I appreciate you doing my work for me

File: data/json/martialarts.json [hunk 1]
```diff
@@ -511,6 +511,8 @@
       "shishkebab_off",
       "shishkebab_on",
       "spear_knife",
+      "spear_knife_superior",
+      "spike_spike"
```
Comment (LaVeyanFiend) [inline data/json/martialarts.json]:
```suggestion
      "spike_spike",
```

Comment (LaVeyanFiend) [reply data/json/martialarts.json]: I forgot me comma :P

File: data/json/martialarts.json [hunk 2]
```diff
@@ -511,6 +511,8 @@
       "shishkebab_off",
       "shishkebab_on",
       "spear_knife",
+      "spear_knife_superior",
+      "spike_spike",
```
Comment (ZhilkinSerg) [inline data/json/martialarts.json]:
```suggestion
      "spear_spike",
```

Comment (Mecares) [reply data/json/martialarts.json]: Damn spiked spikes? I was still using regular spikes on my deathmobile.

Comment (LaVeyanFiend) [reply data/json/martialarts.json]: It's spikes that split off into even more spikes of course ;)

**Pull request merged**